string output_file
bool is_full
---
bool success
string msg